### ISSUE TITLE
HLS type not working on iOS browser

### ISSUE THREAD

**Issue by joshephan** [cookpete/react-player#1685]
Hello, I've checked almost everything, tested every scenario, but it doesn't work, so I leave an issue like this.

#### Current Behavior
1. Access the address on the public cloud front to call up HLS type images. 
2. Works fine in desktop environments and Android. 
3. If you key the forceHls setting in the file object in config, it works fine when you change the User Agent in the browser to the iPhone on your desktop.
4. However, it does not work in a real iPhone environment. 
5. We tried all sorts of things by reflecting most of the solutions registered in the repository issue. However, it was unsuccessful.

#### Steps to Reproduce
```
<ReactPlayer
    className='react-player'
    playsinline
    ref={playerRef}
    width='100%'
    height='100%'
    url={videoUrl}
    playing={player.playing}
    volume={player.volume}
    muted={player.muted}
    playbackRate={player.playerRate}
    pip={player.pip}
    config={{
        file: {
          forceSafariHLS: true,
          forceVideo: true,
          hlsVersion: '1.4.12', // latest hlsversion
          hlsOptions: {
            xhrSetup: function (xhr: any, url: any) {
              xhr.setRequestHeader('token', 'admin');
            }
          }
        }
      }}
/>
```

#### Environment
- URL attempting to play: https://d2ous6lm13gwvv.cloudfront.net/streaming/221118110645645_55e89475-cab1-4bc2-b53c-1785c8d33385/HLS/221118110645645_55e89475-cab1-4bc2-b53c-1785c8d33385.m3u8
- Browser: iPhone Safari and iPhone browser apps(like Google browser)
- Operating system: iOs

**Comment by joshephan** [comment]
After six hours of searching the problem, we came to the following conclusion.

In an iOS environment, HLS is built in, so you have to do something like 

`forceHLS: !isSafari`

and at the same time, hlsOptions won't work.

In my case, 

`xhr.setRequestHeader ('token', 'admin')` in hlsOptions; I couldn't access streaming data in my mobile environment because if I didn't send the command, cloudfront would return a 401 error.

To solve this problem, fetch sent an xhr request by putting a header in the video address to bind the returned value to the url value. But when I looked it up, I found that it's not available for streaming data.

```
useEffect(() => {
        fetch(videoUrl, options)
        .then(response => response.blob())
            .then(blob => {
            setUrl(URL.createObjectURL(blob))
            
        });
    }, [videoUrl])

```

The bottom line is that I think my problem will be solved only if I import streaming data without having to set or set the xhr header in hlsOptions. Please help if anyone has solved the problem in this regard.

**Comment by joshephan** [comment]
After six hours of searching the problem, we came to the following conclusion.

In an iOS environment, HLS is built in, so you have to do something like 

`forceHLS: !isSafari`

and at the same time, hlsOptions won't work.

In my case, 

`xhr.setRequestHeader ('token', 'admin')` in hlsOptions; I couldn't access streaming data in my mobile environment because if I didn't send the command, cloudfront would return a 401 error.

To solve this problem, fetch sent an xhr request by putting a header in the video address to bind the returned value to the url value. But when I looked it up, I found that it's not available for streaming data.

```
useEffect(() => {
        fetch(videoUrl, options)
        .then(response => response.blob())
            .then(blob => {
            setUrl(URL.createObjectURL(blob))
            
        });
    }, [videoUrl])

```

The bottom line is that I think my problem will be solved only if I import streaming data without having to set or set the xhr header in hlsOptions. Please help if anyone has solved the problem in this regard.

**Comment by joshephan** [comment]
Share the test set.

[https://cookpete.com/react-player/](https://cookpete.com/react-player/)

**Custom url**: https://artrooms-stg.s3.ap-northeast-2.amazonaws.com/streaming/2209140444154415_cdff0bc2-9e8d-4f47-9bbe-a760a8767f3e/HLS/2209140444154415_cdff0bc2-9e8d-4f47-9bbe-a760a8767f3e_1080.m3u8

If you change the user agent, including Chrome, to an iPhone-related device, it returns an onError to the console.

**Comment by joshephan** [comment]
I were able to resolve 401 errors that occurred across the cloud front by connecting directly to s3 in an iOS environment. The network history also receives a normal response, so the mobile development tool does not catch any errors, but still does not output images. If you click on the video tag, it appears that the src value is missing as follows.

<img width="403" alt="스크린샷 2023-10-05 오후 2 02 14" src="https://github.com/cookpete/react-player/assets/4365470/9267dfde-0877-492e-b828-1f30fdfec9c4">
<img width="606" alt="스크린샷 2023-10-05 오후 2 02 39" src="https://github.com/cookpete/react-player/assets/4365470/7f18c2f5-d637-4e01-b67a-3e6c04fb7880">

**Comment by joshephan** [comment]
Self solved...

The two problems were mixed and it didn't go well. Another 401 address error that occurs when calling up an address in cloudfront is a lack of understanding of how HLS options work in iPhone environments.

JS code like this

```
const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);

<ReactPlayer
  ...props
  url={
    isSafari ? 'S3_URL' : 'CLOUDFRONT_URL'
  }
  config={{
    file: {
      forceSafariHLS: !isSafari,
      forceVideo: true,
      // in iOS env: hls options is not working 
      hlsOptions: {
        xhrSetup: function (xhr: any, url: any) {
          xhr.setRequestHeader('token', 'admin');
        }
      },
  }}
/>
```

When I wasn't sure how HLS works, I tried to fix the problem by setting it to forceSafariHLS: true, but when I changed the User Agent in my desktop environment and tested it in my iOS environment, it might look like it's working.
